internal/parser/parser.go

```diff
@@ -8910,6 +8910,9 @@ func (p *parser) exprCanBeRemovedIfUnused(expr ast.Expr) bool {
 		*ast.EString, *ast.EThis, *ast.ERegExp, *ast.EFunction, *ast.EArrow, *ast.EImportMeta:
 		return true
 
+	case *ast.EBinary:
```

Comment:
I'm not very familiar with Go, but I would assume order might have an effect on performance in terms of relative frequency of these nodes in an average AST.

I just added this to the top of the non-trivial special cases based on alphabetical order, but this might be something to consider.

Reply:
Speaking of performance, obviously any extra minification optimization doesn't come for free. I have not yet done any profiling, but I noticed there is already a similar optimizations for array literals, so I figured this was probably just an oversight rather than an intentional omission.